internal/pkg/cmd/prerunner.go

```diff
@@ -281,8 +269,10 @@ func (r *PreRun) Authenticated(command *AuthenticatedCLICommand) func(*cobra.Com
 			return err
 		}
 
-		if err := r.ValidateToken(command.Config, unsafeTrace); err != nil {
-			return err
+		if err := r.ValidateToken(command.Config); err != nil {
+			if err := r.updateToken(err, command.Config.Context(), unsafeTrace); err != nil {
```

Comment:
It's slightly awkward to read this line w/ two different `err` variables on the same line; how about using `err2` for the inner one?